File: 1029_two_city_scheduling/two_city_sched_cost.py
from typing import List


def two_city_sched_cost(costs: List[List[int]]) -> int:
    """
    Time: (O n * log(n))
    Space: O(1)
    """
    costs.sort(key=lambda x: x[0] - x[1])
    total = 0
    n = len(costs) // 2
    for i in range(n):
        total += costs[i][0] + costs[i + n][1]
    return total

File: 1029_two_city_scheduling/two_city_sched_cost.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

/**
 * Time: (O n * log(n))
 * Space: O(1)
 */
int two_city_sched_cost(vector<vector<int>> &costs)
{
    sort(begin(costs), end(costs),
         [](const vector<int> &o1, const vector<int> &o2) {
             return (o1[0] - o1[1] < o2[0] - o2[1]);
         });
    int total = 0;
    int n = costs.size() / 2;
    for (int i = 0; i < n; ++i)
        total += costs[i][0] + costs[i + n][1];
    return total;
}
